src/compile/data/bin.ts

```diff
@@ -89,6 +90,27 @@ export class BinNode extends DataFlowNode {
     return new BinNode(bins);
   }
 
+  public static makeBinFromTransform(model: Model, t: BinTransform) {
+    const bins: Dict<BinComponent> = {};
+
+    if (t.bin) {
+      const bin: Bin = isBoolean(t.bin) ? {} : t.bin;
+      const key = `${binToString(t.bin)}_${t.field}`;
+      bins[key] = {
```

Comment:
You might want to extract a method so that we use common method to create the object here.